### ISSUE TITLE
[FEAT] : Cart Slice

### ISSUE THREAD

**Issue by Sid-80** [pooranjoyb/popShop#94]
## Description
Adding the cart slice to manage the state of cart

**Comment by pooranjoyb** [comment]
Congratulations, @Sid-80! 🎉 Thank you for creating your issue. Your contribution is greatly appreciated and we look forward to working with you to resolve the issue. Keep up the great work!<p>We will promptly review your changes and offer feedback. Keep up the excellent work! Kindly remember to check our [contributing guidelines](https://github.com/pooranjoyb/popShop/blob/dev/CONTRIBUTING.md)</p>

**Comment by Mahitej28** [comment]
@Sid-80 Can you elaborate more on your issue.

**Comment by Sid-80** [comment]
@Mahitej28 
In future after adding the functionality of adding item to the cart, there will be need of storing the state of items present in the cart. So, here we can use redux.
So, I want to implement the cart slice for storing the item.

**Comment by Mahitej28** [comment]
@pooranjoyb what are your thoughts on this?

**Comment by pooranjoyb** [comment]
Hiii @Sid-80 as of now, we wanna put this issue for hold. As most of the cart items will be fetched from the supabase itself.

**Comment by masterboy376** [comment]
Hello @pooranjoyb I would like to work on this one under GSSOC 24

**Comment by pooranjoyb** [comment]
Go ahead 🚀

**Comment by Sid-80** [comment]
@pooranjoyb I made the request earlier

**Comment by Mahitej28** [comment]
@Sid-80 give your updates by EOD

**Comment by Sid-80** [comment]
I am travelling back to home today, will start working by tomorrow

**Comment by Sid-80** [comment]
@Mahitej28 @pooranjoyb Hey I have created the slice for cart and added reducers to it!
And everything is static, so I am just able to make the number present on the cart at Navbar as dynamic

**Comment by Sid-80** [comment]
![image](https://github.com/pooranjoyb/popShop/assets/101277141/2aca017d-b498-4ed9-990b-a0382f911403)

**Comment by Sid-80** [comment]
![image](https://github.com/pooranjoyb/popShop/assets/101277141/22c9107c-d4eb-4681-ba16-d6d62be88f46)

**Comment by Sid-80** [comment]
@pooranjoyb @Mahitej28 Let me know if any changes are required!

**Comment by Sid-80** [comment]
@pooranjoyb  Yes I have created array containing the item data with unique id

**Comment by Sid-80** [comment]
https://github.com/pooranjoyb/popShop/issues/94#issuecomment-2143509818
 I have created ITEM interface for it

**Comment by pooranjoyb** [comment]
> [#94 (comment)](https://github.com/pooranjoyb/popShop/issues/94#issuecomment-2143509818) I have created ITEM interface for it

So, as of now it's just the slice right?
We are not updating the state on adding item to slice right? or we doing that too? I mean it's functionality? @Sid-80

**Comment by Sid-80** [comment]
Nope I have added the functions but due to static data it's not possible!

**Comment by pooranjoyb** [comment]
> Nope I have added the functions but due to static data it's not possible!
 
Cool. That'll do. Raise a PR for the same. @Sid-80

**Comment by Sid-80** [comment]
Sure!
